src/identity-providers/add/AddMapper.tsx

```diff
@@ -372,92 +382,201 @@ export const AddMapper = () => {
+                          setTargetOptionsOpen(false);
+                        }}
+                        selections={value}
+                        variant={SelectVariant.single}
+                        aria-label={t("target")}
+                        isOpen={targetOptionsOpen}
+                      >
+                        {targetOptions.map((option) => (
+                          <SelectOption
+                            selected={option === value}
+                            key={option}
+                            data-testid={option}
+                            value={option}
+                          >
+                            {t(`targetOptions.${option}`)}
+                          </SelectOption>
+                        ))}
+                      </Select>
+                    )}
+                  />
+                </FormGroup>
+              </>
+            )}
+            {(form.getValues().identityProviderMapper ===
+              "saml-advanced-role-idp-mapper" ||
+              form.getValues().identityProviderMapper ===
+                "oidc-hardcoded-role-idp-mapper" ||
+              form.getValues().identityProviderMapper ===
+                "saml-role-idp-mapper") && (
+              <FormGroup
+                label={t("common:role")}
+                labelIcon={
+                  <HelpItem
+                    id="name-help-icon"
+                    helpText="identity-providers-help:role"
+                    forLabel={t("identity-providers-help:role")}
+                    forID={t(`identity-providers:helpLabel`, {
+                      label: t("role"),
+                    })}
+                  />
+                }
+                fieldId="kc-role"
                 validated={
                   errors.config?.role
                     ? ValidatedOptions.error
                     : ValidatedOptions.default
                 }
-              />
-              <Button
-                data-testid="select-role-button"
-                onClick={() => toggleModal()}
+                helperTextInvalid={t("common:required")}
               >
-                {t("selectRole")}
-              </Button>
-            </FormGroup>{" "}
+                <TextInput
+                  ref={register()}
+                  type="text"
+                  id="kc-role"
+                  data-testid="mapper-role-input"
+                  name="config.role"
+                  value={selectedRole[0]?.name}
+                  validated={
+                    errors.config?.role
+                      ? ValidatedOptions.error
+                      : ValidatedOptions.default
+                  }
+                />
+                <Button
+                  data-testid="select-role-button"
+                  onClick={() => toggleModal()}
+                >
+                  {t("selectRole")}
+                </Button>
+              </FormGroup>
+            )}{" "}
```

Comment:
Perhaps this white space can be removed?
```suggestion
            )}
```